schema/2.0.0/corpora_schema.md

```diff
@@ -0,0 +1,257 @@
+| [ENSEMBL (Mouse)] | Mouse reference mm10 (GENCODE vM23/Ensembl 98) matching [cellranger 2020-A (July 7, 2020) release]| [gencode.vM23.primary_assembly.annotation.gtf] |
+|||
+
+[ENSEMBL (Human)]: http://www.ensembl.org/Homo_sapiens/Info/Index
+[gencode.v32.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.primary_assembly.annotation.gtf.gz"
+
+[ENSEMBL (Mouse)]: http://www.ensembl.org/Mus_musculus/Info/Index
+[gencode.vM23.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M23/gencode.vM23.primary_assembly.annotation.gtf.gz"
+
+[cellranger 2020-A (July 7, 2020) release]: https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build
+
+## `obs` (Cell Metadata)
+
+`obs` is a [`pandas.DataFrame`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html).
+
+Curators MUST annotate the following columns in the `obs` dataframe:
+
+| Key | Value | Annotator |
+|-|-|-|
+| assay_ontology_term_id | categorical with `str` categories. This MUST be a child of "experimental process". See [EFO:0002694](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0002694) for details. If there is not an exact match for the assay, *clarifying text* MAY be appended in parentheses such as `" (sci-plex)"`. | Curator |
+| cell_type_ontology_term_id | categorical with `str` categories. This MUST be a CL term. | Curator |
+| development_stage_ontology_term_id | categorical with `str` categories. If unavailable, this MUST be `"unknown"`. <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:9606"`, this MUST be a HsapDv term.<br><br> If `organism_ontolology_term_id` is `"NCBITaxon:10090"`, this MUST a MmusDv term with the following restrictions: <br><br> **Prenatal stages** MUST be in the range beginning with "MmusDv:0000003" and ending with "MmusDv:0000035". <br><br> **Postnatal stages (1-4 weeks)** MUST be in the range beginning with "MmusDv:0000045" and ending with "MmusDv:0000048". <br><br> **Postnatal stages (after week 4)** MUST be in the range beginning with "MmusDv:0000062" and ending with "MmusDv:0000091". | Curator |
+| disease_ontology_term_id | categorical with `str` categories. This MUST be a MONDO term or [PATO:0000461](http://purl.obolibrary.org/obo/PATO_0000461) for "normal". | Curator |
+| ethnicity_ontology_term_id | categorical with `str` categories. If `organism_ontolology_term_id` is `"NCBITaxon:9606"`, this MUST be either a HANCESTRO term or `"unknown"` if unavailable. <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:10090"`, this MUST be `"na"`. | Curator |
+| is_primary_data | `bool`. This MUST be `True` if this is the canonical instance of this cellular observation and `False` if not. This is commonly `False` for meta-analyses reusing data or for secondary views of data. | Curator |
```

Comment:
> Would the submitter be expected to flag each cell as 'primary' exactly once across those?

Yes, they'd be expected to decide which view of each cell they wanted to mark as primary. For simplicity, I'd suggest the "all cells" view.

Reply:
Will they be a problem when 2 researchers each do their integrative analysis which includes a dataset from ArrayExpress that is not submitted to cellxgene? Those cells would appear in multiple collections/datasets on the Data Portal, but would not be marked as 'primary' in any case.

Reply:
The way we've solved that problem in the past is by asking the researcher to submit the original dataset on behalf of the lab whose data they're reusing. We have a dataset that will shortly be published that did this for all 5 datasets they used. 

That said, this might be something we need to resolve, yes. Mark one as primary and one as secondary. Do you see a better solution? It's important that our software only count datasets once when doing integrative visualizations.

Reply:
I had been wondering how you were going to avoid duplication once the integrations start - seems like a great solution to me. That info would be good in here somewhere. "If this is not the canonical instance of this observation and that instance has not already been submitted to the Data Portal..."
Curious how you'll validate each cell is 'primary' once, and no cell is 'primary' multiple times.

Reply:
@ambrosejcarr - ^^^^ ?

Reply:
There are two options. For now I suggest option 2, since option 1 would be challenging to implement: 

1. Check if the raw cell expression vector has been submitted before. My intuition is that the probability of getting two identical expression vectors is very low, even into the billions of cells. We could run the Poisson statistics to check, of course. 
2. It's part of lattice's curation QA check. I think it's reasonable that you could catch the majority of cases because within-experiment duplication should be fairly obvious, and reuse of other previously submitted data should also be detectable because reusing someone else's data without citing them is very bad form. 

What do you think?

Reply:
2 seems fine.
We have questions about how to go about submitting datasets as primary in order to allow the submission of the integration dataset anyway, so I think my other questions will be answered in that discussion